src/stdune/path.ml

```diff
@@ -511,7 +511,7 @@ end = struct
     | _               , In_source_tree _ -> Gt
     | In_build_dir x  , In_build_dir y   -> Local.compare x y
 
-  let equal x y = compare x y = Ordering.Eq
+  let equal = (=)
```

Comment:
We should always use a type annotation when rebinding a polymorphic comparison function. This way the compiler might generate specialized code.